### DISCUSSION TITLE
Using reload to modify global subscriber once it's already been set

### DISCUSSION THREAD

**Discussion by vamshiaruru-virgodesigns** [tokio-rs/tracing#2454] [Q&A]
Hi everyone, I need some help with setting up tracing in a little niche scenario. I am trying to write a logging library that can be used from python as a practice project. I'd like to expose a method where in the python code, the library user can do something like
```python
logger1 = logging.getLogger("logger1")
logger2 = logging.getLogger("logger2")
# logger1, and logger2 can have their sinks, formatters, filters etc.
```
My idea to implement this in rust would be to have separate writers for each logger with their own formatters and what not. Every time a `getLogger` is called, and formatter/sinks are added on the logger, I'd use `reload` to update the global subscriber and set new writers and filters. I am a complete beginner, so my question is  am I thinking in the right direction? Is there an easier way of doing this? Is it possible to just reset the global subscriber completely and build it from scratch every time, and would that be better?

Any help is very appreciated. Thank you :)

**Comment by davidbarsky** [comment]
Depending on how often you expect a user to call `getLogger` with new arguments, calling `reload` to update the global subscriber is probably reasonable. Note that there's a non-trivial performance impact from calling `Reload`, in that all spans and events need to be re-evaluated for interest, but if you don't expect that to occur frequently, then this approach is completely reasonable!
